### DISCUSSION TITLE
Vim motion extension stops processing the keyboard input

### DISCUSSION THREAD

**Discussion by GersonDias** [microsoft/vscode-discussions#2430] [Extension Development QnA] [answered]
Hello all,

I'm using [amVim extension](https://github.com/aioutecism/amVim-for-VSCode) and, sometimes, when I'm holding a motion key (for instance 'k') the cursor starts going up, but suddenly the row is deleted and the key I'm holding is inserted. Looks like the extension just stop processing the command and the vscode handles that input.

A short video showing the behaviour:

https://github.com/user-attachments/assets/6ff16b6a-0b75-4b1f-9a14-33bb4c1d2444

Looks like it happens more right after I save the file (formatting on save is enabled). and also my project is quite huge. And I started notice it a few weeks ago. So I'm not sure if this is a vscode problem or a problem with that extension... Anyone notice something similar?

**Comment by alefragnani** [comment, marked as answer]
Hi @GersonDias ,

If you are having issues using the **amVim** extension, the best place to look for would be in the extension's repo / issues.

I'm not sure how the extension works, but it seems other users have reported intermittent cursor movement as well https://github.com/aioutecism/amVim-for-VSCode/issues/308. Maybe it is the same error you are facing. I would suggest you to look on this issue, if the error is the same, add comments on your scenario. Or maybe, open a new issue. 

If the amVim extension team/developer discovers the issue is on VS Code, it would contact the VS Code team reporting the issue. Or they could ask for help here, as this discussion forum is for extension development.

Hope this helps

**Comment by GersonDias** [comment]
Hi @alefragnani,

Thank you for your response!

I actually opened an [issue](https://github.com/aioutecism/amVim-for-VSCode/issues/334) in the extension's repository, but unfortunately, I haven’t received any replies yet. I’m hoping to hear back soon. That said, the issue you pointed out doesn’t seem to align with the one I’m experiencing.

By the way, I’ve noticed that the problem tends to occur more frequently when I have some terminal sessions open within VS Code. Is this a common challenge in VS Code extension development?

Thanks again for your input!
